main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -87,6 +87,14 @@ object Typer {
       }.toSet
       val syms = syms0 ++ namespaces
 
+      val constructs_with_companion_module = traits.keys ++ enums.keys ++ structs.keys ++ effects.keys
+      // The companion modules, empty by default.
+      val companion_modules =
+        for {
+          sym <- constructs_with_companion_module
+          modSym = new Symbol.ModuleSym(sym.namespace :+ sym.name, ModuleKind.Companion)
+        } yield modSym -> List.empty[Symbol]
```

Comment:
I think this whole for-thing is complicated. Can we not do a simple fold or map?